### ISSUE TITLE
CUDA out of memory. 16GB VRAM. RTX 4070Ti SUPER

### ISSUE THREAD

**Issue by aesxsc** [kinggongzilla/ai-clone-whatsapp#12]
LLama3 
`torch.OutOfMemoryError: CUDA out of memory. Tried to allocate 628.00 MiB. GPU 0 has a total capacity of 15.60 GiB of which 107.62 MiB is free. Including non-PyTorch memory, this process has 15.47 GiB memory in use. Of the allocated memory 14.56 GiB is allocated by PyTorch, and 675.27 MiB is reserved by PyTorch but unallocated. If reserved but unallocated memory is large try setting PYTORCH_CUDA_ALLOC_CONF=expandable_segments:True to avoid fragmentation.  See documentation for Memory Management  (https://pytorch.org/docs/stable/notes/cuda.html#environment-variables)
`
Are there any ways to run with 16GB of VRAM?

**Comment by aesxsc** [comment]
I enabled "enable_activation_checkpointing: False" to True and now it somehow works?

**Comment by aesxsc** [comment]
No, it still fails halfway into it. `torch.OutOfMemoryError: CUDA out of memory. Tried to allocate 1.96 GiB. GPU 0 has a total capacity of 15.60 GiB of which 1.70 GiB is free. Including non-PyTorch memory, this process has 13.87 GiB memory in use. Of the allocated memory 12.76 GiB is allocated by PyTorch, and 871.76 MiB is reserved by PyTorch but unallocated. If reserved but unallocated memory is large try setting PYTORCH_CUDA_ALLOC_CONF=expandable_segments:True to avoid fragmentation.  See documentation for Memory Management  (https://pytorch.org/docs/stable/notes/cuda.html#environment-variables)
1|853|Loss: 0.7853811383247375:  63%|████████████████████████              | 853/1347 [23:19<13:30,  1.64s/it]`

**Comment by neooriginal** [comment]
same issue, did you find a fix?
